internal/cephfs/store/volumeoptions.go

```diff
@@ -542,9 +542,10 @@ func (vo *VolumeOptions) populateVolumeOptionsFromBackingSnapshot(
 		return fmtBackingSnapshotOptionMismatch("clusterID", vo.ClusterID, parentBackingSnapVolOpts.ClusterID)
 	}
 
-	if vo.Pool != "" {
-		return errors.New("cannot set pool for snapshot-backed volume")
-	}
+	// Pool parameters is optional and only used to set 'pool_layout' argument for
+	// subvolume and subvolume clone create commands.
+	// Setting this to empty since it is not used with Snapshot-backed volume.
+	vo.Pool = ""
```

Comment:
Why should this not be the same as the pool of the original volume?

Reply:
> Why should this not be the same as the pool of the original volume?

The original volume's pool is set only for [create pvc/clone cmd ](https://github.com/search?q=repo%3Aceph/ceph-csi%20PoolLayout&type=code) as `pool_layout` (pool is retrieved directly from pv attributes) and is not saved anywhere and not used anywhere else.

The `parentBackingSnapVolOpts` here does not have its pool parameter set.